### PR TITLE
Remove conditional lookups

### PR DESCRIPTION
**Pull request by ptomato**
This addresses Richard's comments on #2519.

Tests are in https://github.com/tc39/test262/pull/3959

Closes: #2724

### CODE REVIEW

**Review by justingrant — APPROVED** [review]
👍

File: spec/duration.html [hunk 1]
```diff
@@ -105,9 +105,7 @@ <h1>Temporal.Duration.compare ( _one_, _two_ [ , _options_ ] )</h1>
         1. Let _calendarRec_ be *undefined*.
         1. If _zonedRelativeTo_ is not *undefined* or _plainRelativeTo_ is not *undefined*, then
           1. If _zonedRelativeTo_ is not *undefined*, let _calendar_ be _zonedRelativeTo_.[[Calendar]]; else let _calendar_ be _plainRelativeTo_.[[Calendar]].
-          1. Set _calendarRec_ to ! CreateCalendarMethodsRecord(_calendar_, « »).
-          1. If _calendarUnitsPresent_ is *true*, then
-            1. Perform ? CalendarMethodsRecordLookup(_calendarRec_, ~dateAdd~).
+          1. Set _calendarRec_ to ? CreateCalendarMethodsRecord(_calendar_, « ~dateAdd~ »).
```
Comment (gibson042) [inline spec/duration.html:108]: I wonder if the four instances of this _`If _zonedRelativeTo_ is not *undefined* or _plainRelativeTo_ is not *undefined*, … Set _calendarRec_ to ? CreateCalendarMethodsRecord(_calendar_, « … »)`_ pattern justify introducing a new operation.
```suggestion
        1. Let _calendarRec_ be CreateCalendarMethodsRecordFromField(« _zonedRelativeTo_, _plainRelativeTo_ », « ~dateAdd~ »).
```

**Review by gibson042 — APPROVED** [review]

File: spec/duration.html (hunk 1 above)
Comment (ptomato) [reply spec/duration.html]: Thanks for the suggestion. Added MaybeCreateCalendarMethodsRecordFromRelativeTo.

**Comment by ptomato** [comment]
@gibson042 Would you be able to check if you agree with https://github.com/tc39/test262/pull/3959 as well, so we can get this merged promptly?

File: spec/calendar.html [hunk 2]
```diff
@@ -186,6 +186,25 @@ <h1>
       </emu-alg>
     </emu-clause>
 
+    <emu-clause id="sec-temporal-maybecreatecalendarmethodsrecordfromrelativeto" type="abstract operation">
+      <h1>
+        MaybeCreateCalendarMethodsRecordFromRelativeTo (
```
Comment (gibson042) [inline spec/calendar.html:191]: I think "Maybe" can be dropped from the name, "CreateCalendarMethodsRecordFromRelativeTo" reads just as well without misleading w.r.t. undefined in my opinion.

File: spec/calendar.html [hunk 3]
```diff
@@ -186,6 +186,25 @@ <h1>
...
+          _methods_: a List of one of ~dateAdd~, ~dateFromFields~, ~dateUntil~, ~day~, ~fields~, ~mergeFields~, ~monthDayFromFields~, or ~yearMonthFromFields~,
+        ): either a normal completion containing either a Calendar Methods Record or *undefined*, or a throw completion
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>It creates a Calendar Methods Record from the calendar of the given `relativeTo` object, looking up the methods given in _methods_. If no `relativeTo` object is given, it returns *undefined* instead of a Calendar Methods Record.</dd>
+      </dl>
+      <emu-alg>
+        1. If _zonedRelativeTo_ is *undefined* and _plainRelativeTo_ is *undefined*, return *undefined*.
+        1. If _zonedRelativeTo_ is not *undefined*, let _calendar_ be _zonedRelativeTo_.[[Calendar]]; else let _calendar_ be _plainRelativeTo_.[[Calendar]].
```
Comment (gibson042) [inline spec/calendar.html:203]: Nit: I think having a dedicated operation (and an alias for the list of field names) allows this logic to better highlight the similar treatment of plain vs. zoned input.
```
        1. If _zonedRelativeTo_ is not *undefined*, return ? CreateCalendarMethodsRecord(_zonedRelativeTo_.[[Calendar]], _methods_).
        1. If _plainRelativeTo_ is not *undefined*, return ? CreateCalendarMethodsRecord(_plainRelativeTo_.[[Calendar]], _methods_).
        1. Return *undefined*.
```

**Review by gibson042 — APPROVED** [review]
> @gibson042 Would you be able to check if you agree with [tc39/test262#3959](https://github.com/tc39/test262/pull/3959) as well, so we can get this merged promptly?

Done.

File: spec/calendar.html (hunk 3 above)
Comment (ptomato) [reply spec/calendar.html:203]: Agreed, I like this better.

**Comment by ptomato** [comment]
Thanks for the reviews.

**Pull request merged**